crates/ironrdp-server/src/lib.rs

```diff
@@ -28,3 +28,23 @@ pub use handler::*;
 pub use helper::*;
 pub use server::*;
 pub use sound::*;
+
+#[cfg(feature = "bench")]
+pub mod bench {
+    pub mod encoder {
+        pub use crate::encoder::*;
+    }
+}
+
+#[macro_export]
+macro_rules! time_warn {
+    ($context:expr, $threshold:expr, $op:expr) => {{
+        let start = std::time::Instant::now();
+        let result = $op;
+        let duration = start.elapsed().as_millis();
+        if duration > $threshold {
+            ::tracing::warn!("{} took {} ms!", $context, duration);
```

Comment:
question: Should we try to backoff in case the server is weak or a regression is shipped?

Example:
```rust
#[macro_export]
macro_rules! time_warn {
    ($context:expr, $threshold:expr, $op:expr) => {{
        #[cold]
        fn warn_log(context: &str, duration: Duration) {
            static COUNT: AtomicUsize = AtomicUsize::new(0);
            let current_count = COUNT.fetch_add(1);
            if current_count < 50 || current_count % 100 == 0 {
                ::tracing::warn!("{context} took {duration} ms! (count: {current_count})");
            }
        }

        let start = std::time::Instant::now();
        let result = $op;
        let duration = start.elapsed().as_millis();
        if duration > $threshold {
            warn_log($context, duration);
        }
        result
    }};
}
```

Generating a unique function per `time_warn!` is intended so the counter is not shared.

Reply:
sounds good to me. There is probably a crate for that already ;)

Reply:
If you find something suitable we can consider using it :slightly_smiling_face:

Reply:
there is https://lib.rs/crates/logging_timer, but it's not quite the same :)